### DISCUSSION TITLE
GenericDataset AttributeError: module 'netCDF4' has no attribute 'Dataset'

### DISCUSSION THREAD

**Discussion by Flash-Of-Thunder** [neuralhydrology/neuralhydrology#55] [Q&A] [answered]
I'm trying to get the GenericDataset function to work with the most basic input features before scaling up, and got the above error. At first I assumed it's something with my netcdf format. But then I recreated a clean environment w/ just the Neural Hydrology download and got the following error: 

ValueError: found the following matches with the input file in xarray's IO backends: ['netcdf4', 'h5netcdf']. But their dependencies may not be installed, see:
http://xarray.pydata.org/en/stable/user-guide/io.html 
http://xarray.pydata.org/en/stable/getting-started-guide/installing.html

``` Code I'm running 
import os
import pickle
from pathlib import Path

import matplotlib.pyplot as plt
from neuralhydrology.evaluation import metrics
from neuralhydrology.nh_run import start_run, eval_run


start_run(config_file=Path("C:/Users/S/Documents/1_basin_generic_2.yml"))
```
[1_basin.txt](https://github.com/neuralhydrology/neuralhydrology/files/7738737/1_basin.txt)

[1_basin_generic_2.zip](https://github.com/neuralhydrology/neuralhydrology/files/7738796/1_basin_generic_2.zip)

[8220000.zip](https://github.com/neuralhydrology/neuralhydrology/files/7738797/8220000.zip)

**Comment by Flash-Of-Thunder** [comment]
I looked into this some more and I was able to get past this error by installing neural hydrology on a clean environment and then installing both dependencies in the terminal with `pip install netcdf4` and `pip install h5netcdf`. Still getting new errors but I think its because the netcdf file I made is missing a coherent time format.

**Comment by kratzert** [comment, marked as answer]
Hi @Flash-Of-Thunder 

it has hard to reason about dependency problems in the Python ecosystem, sorry. Because of this, we provide an environment file so that you should not have these problems. If I understand you correctly, you still haven't created a conda environment using one of our environment files, but rather just installed the neuralhydrology package in a clean environment.

Please, as a first step, create a new environment with `conda env create -f /path/to/environment.yml`, where the path points to one of the environment file in the environment folder of this repository. Make sure to select the correct one depending on your cuda version and/or GPU availability.

Then, inside of this environment, install neuralhydrology as indicated in the documentation. You should not have any troubles then.

For the future, please use the Issue section for bug/error reports.

**Reply by Flash-Of-Thunder** [reply]
Excellent! Yes I'm slowly learning about the Python ecosystem. I'll report any future errors in the issues section.
